team.py

```diff
@@ -1,88 +1,95 @@
+
+    def display_status(self, teamname):
+        try:
+            team = HuntUser.objects.get(name=teamname)
+            ret = "Team: " + team.name
+        except HuntUser.DoesNotExist:
+            ret = "Team does not exist"
+
         return ret
 
-    def edit_password(self, input):
+    # def edit_username(self, teamname, input):
+    #     username = input[0]
+    #
+    #     try:
+    #         team = HuntUser.objects.get(name=teamname)
+    #         team.name = username
+    #         team.save()
+    #         ret = "Username successfully changed to " + username
+    #     except HuntUser.DoesNotExist:
+    #         ret = "Team does not exist"
+    #
+    #     ret = "Username successfully changed to " + username
+    #     return ret
+
+
+    def edit_password(self, teamname, input):
         password = input[0]
-        if self.database.get_current_user() is not self:
-            return False
-        self.password = password
-        ret = "Password successfully changed to " + password
+
+        try:
+            team = HuntUser.objects.get(name=teamname)
+            team.password = password
+            team.save()
+            ret = "Password successfully changed to " + password
+        except HuntUser.DoesNotExist:
+            ret = "Team does not exist"
+
         return ret
 
-    def answer_question(self, input):
-        user_answer = input[0]
-        if self.database.get_current_user() is not self:
-            return "Cannot answer question when not logged in!"
-        if not self.database.game_running:
-            return "There is no current game to answer a question for"
-        landmarks = self.database.get_landmark_path()
+    def answer_question(self, teamname, input):
+        answer = input[0]
+        game = Game.objects.get(name="game")
+        if not game.running:
+            return "There is no game running!"
         try:
-            check_landmark = landmarks[self.current_landmark+1]
-        except IndexError:
-            return "Not at a valid landmark"
-        if check_landmark.verify_answer(user_answer):
-            self.current_landmark += 1
-            if self.current_landmark == (len(landmarks)-1):
-                self.database.game_running = False
-                return "Congratulations! Your team has won the game!"
-            return "Correct answer given! You can now request the clue for the next landmark"
-        else:
-            self.penalties += 1
-            return "Incorrect answer, please try again"
+            team = HuntUser.objects.get(name=teamname)
+        except HuntUser.DoesNotExist:
+            return "Not a valid team!"
 
+        if answer != team.current_landmark.answer:
             ret_string = "Incorrect answer, please try again"
+        else:
+            try:
+                next_landmark = Landmark.objects.get(order_num=team.current_landmark.order_num + 1)
+            except Landmark.DoesNotExist:
```

Comment:
Maybe add another parameter that keeps track of the last landmark? Using an try-except block as the winning condition seems improper to me.

Reply:
"add another parameter" doesn't really make sense to me. Do you mean storing that in the database?

Using get within a try block is the standard way I found online for anything regarding Django's get method.